planner/core/logical_plans.go

```diff
@@ -631,6 +631,12 @@ func isColEqCorColOrConstant(filter expression.Expression, col *expression.Colum
 
 func (ds *DataSource) getPKIsHandleCol() *expression.Column {
 	if !ds.tableInfo.PKIsHandle {
+		// If the PKIsHandle is false, return the ExtraHandleColumn.
+		for i, col := range ds.Columns {
```

Comment:
I guess finding column by ID should be a very common pattern, can we define it in `model.go` so others can use it?

Reply:
Agreed. @winoros How do you think?

Reply:
It might be a little wired. Here, we want to find the index `i` of the column in `[]*ColumnInfo`, so that we can use `i` to return a `*Column`. If we define the `FindColumnInfoByID`, it may look like this :
```go
func FindColumnInfoByID(cols []*ColumnInfo, id int64) （int, *ColumnInfo) {
...
} 
```
Is this ok?

Reply:
Oh, no need to change it then.

Reply:
Sure, i put it in planner just because that i don't need to modify parser then i can resolve the failed tests in my code quickly.